src/routes/game/components/elements/cardPopUp/CardPopUp.tsx

```diff
@@ -1,71 +1,59 @@
-import React from 'react';
-import { useAppSelector } from 'app/Hooks';
-import { RootState } from 'app/Store';
-import CardImage from '../cardImage/CardImage';
-import styles from './CardPopUp.module.css';
-import { doubleFacedCardsMappings } from './constants';
-import classNames from 'classnames';
-import useWindowDimensions from 'hooks/useWindowDimensions';
-
-const popUpGap = 10;
-
-function CardPopUp({ src, containerClass, containerStyle }: { src: string, containerClass?: string, containerStyle?: Record<string, string> }) {
-  const containerClassName = containerClass != null ? containerClass : classNames(styles.defaultPos, styles.popUp);
-  return (
-    <div className={containerClassName} style={containerStyle}>
-        <div className={styles.popUpInside} key={src}>
-        <CardImage src={src} className={styles.img} />
-      </div>
-    </div>);
-}
-
-function getSrcs(cardNumber: string): Array<string> {
-  const cardNumbers = [cardNumber];
-  if (doubleFacedCardsMappings[cardNumber] != null) {
-    cardNumbers.push(doubleFacedCardsMappings[cardNumber]);
-  }
-  return cardNumbers.map((currentCardNumber) => 
-    `/cardimages/${currentCardNumber}.webp`
-  );
-}
-
-export default function CardPopUpContainer() {
-  const popup = useAppSelector((state: RootState) => state.game.popup);
-  const [windowWidth, windowHeight] = useWindowDimensions();
-  if (
-    popup === undefined ||
-    popup.popupOn === false ||
-    popup.popupCard === undefined
-  ) {
-    return null;
-  }
-
-  const [src, dfcSrc] = getSrcs(popup.popupCard.cardNumber);
-
-  if (popup.xCoord === undefined || popup.yCoord === undefined) {
-    return (<CardPopUp src={src} />);
-  }
-
-  const popUpStyle: Record<string, string> = {};
-
-  if (popup.xCoord > windowWidth / 2) {
-    popUpStyle.right =
-      (windowWidth - (popup.xCoord - popUpGap)).toString() + 'px';
-  } else {
-    popUpStyle.left = (popup.xCoord + popUpGap).toString() + 'px';
-  }
-
-  if (popup.yCoord < windowHeight / 2) {
-    popUpStyle.top = popup.yCoord.toString() + 'px';
-  } else {
-    popUpStyle.bottom =
-      (windowHeight - popup.yCoord + popUpGap).toString() + 'px';
-  }
-
-  return (
-    <div className={styles.popUp} style={popUpStyle}>
-      {dfcSrc != null && <CardPopUp src={dfcSrc} containerClass={classNames(styles.popUp, styles.doubleFacedCard)} />}
-      <CardPopUp src={src} containerClass={styles.popUp} containerStyle={popUpStyle} />
-    </div>
-   )
-}
+import { useAppDispatch } from "app/Hooks";
+import { clearPopUp, setPopUp } from "features/game/GameSlice";
+import useWindowDimensions from "hooks/useWindowDimensions";
+import { ReactNode, useRef } from "react";
+
+type CardPopUpProps = { 
+    children: ReactNode, 
+    cardNumber: string, 
+    containerClass?: string, 
+    onClick?: () => void, 
+    isHidden?: boolean 
+};
+
+export default function CardPopUp({ children, cardNumber, containerClass, onClick, isHidden }: CardPopUpProps) {
```

Comment:
This is the component that's replaced a lot of the code. We just need a cardNumber and a child component. When wrapping the child whenever it is hovered we will see the card projected into the card portal.

e.g.
```
<CardPopUp cardNumber="UPR001">
   <span>Dromai</span>
</CardPopUp>
```